integration/shared/isolated/create_service_command_test.go

```diff
@@ -0,0 +1,339 @@
+							"my-service",
+							service,
+							servicePlan,
+						))
+					})
+				})
+
+				When("creating with valid params json in a file", func() {
+					var tempFile *os.File
+
+					BeforeEach(func() {
+						tempFile = newTempFile(`{"valid":"json"}`)
+					})
+
+					AfterEach(func() {
+						Expect(os.Remove(tempFile.Name())).To(Succeed())
+					})
+
+					It("displays an informative success message, exits 0", func() {
+						session := helpers.CF("create-service", service, servicePlan, "my-service", "-c", tempFile.Name())
+						Eventually(session).Should(Say("Creating service instance %s in org %s / space %s as %s\\.\\.\\.",
+							"my-service", org, space, username))
+						Eventually(session).Should(Say("OK"))
+						Eventually(session).Should(Exit(0))
+
+						session = helpers.CF("services")
+						Eventually(session).Should(Exit(0))
+						Eventually(session).Should(Say("%s\\s+%s\\s+%s\\s+create succeeded",
+							"my-service",
+							service,
+							servicePlan,
+						))
+					})
+				})
+
+				When("creating with tags", func() {
+					It("displays an informative message, exits 0, and creates the service with tags", func() {
+						session := helpers.CF("create-service", service, servicePlan, "my-service", "-t", "sapi, rocks")
+						Eventually(session).Should(Say("Creating service instance %s in org %s / space %s as %s\\.\\.\\.",
+							"my-service", org, space, username))
+						Eventually(session).Should(Say("OK"))
+						Eventually(session).Should(Exit(0))
+
+						session = helpers.CF("service", "my-service")
+						Eventually(session).Should(Exit(0))
+						Eventually(session).Should(Say("tags:\\s+sapi, rocks"))
+					})
+				})
+			})
+
+			When("the service provided is async and accessible", func() {
+				var (
+					service     string
+					servicePlan string
+					broker      helpers.ServiceBroker
+				)
+
+				BeforeEach(func() {
+					service = helpers.PrefixedRandomName("SERVICE")
+					servicePlan = helpers.PrefixedRandomName("SERVICE-PLAN")
+
+					broker = helpers.NewAsynchServiceBroker(helpers.NewServiceBrokerName(), helpers.NewAssets().ServiceBroker, domain, service, servicePlan)
+					broker.Push()
+					broker.Configure(true)
+					broker.Create()
+					Eventually(helpers.CF("enable-service-access", service)).Should(Exit(0))
+				})
+
+				AfterEach(func() {
+					broker.Destroy()
+				})
+
+				It("creates the service and displays a message that creation is in progress", func() {
+					session := helpers.CF("create-service", service, servicePlan, "my-service", "-v")
+					Eventually(session).Should(Say("Creating service instance %s in org %s / space %s as %s\\.\\.\\.",
+						"my-service", org, space, username))
+					Eventually(session).Should(Say("OK"))
+					Eventually(session).Should(Say("Create in progress. Use 'cf services' or 'cf service my-service' to check operation status."))
+					Eventually(session).Should(Exit(0))
+				})
+			})
+		})
+	})
+})
+
+func newTempFile(contents string) *os.File {
```

Comment:
This function looks generic enough for it to be added to the helpers package. Possibly in `integration/helpers/file.go`.  Also naming the function more generic like `TempFileWithContent` would make it easier to find.